src/event-handler/receive.ts

```diff
@@ -1,20 +1,19 @@
-"use strict";
-
-module.exports = receiverHandle;
-
-const wrapErrorHandler = require("./wrap-error-handler");
+import { wrapErrorHandler } from "./wrap-error-handler";
+import { WebhookEvent, EventState } from "../index.d";
 
 // main handler function
-function receiverHandle(state, event) {
+export function receiverHandle(state: EventState, event: WebhookEvent) {
   const errorHandlers = state.hooks.error || [];
 
   if (event instanceof Error) {
-    errorHandlers.forEach((handler) => wrapErrorHandler(handler, event));
+    errorHandlers.forEach((handler: Function) =>
+      wrapErrorHandler(handler, event)
+    );
 
     return Promise.reject(event);
   }
 
-  if (!event || !event.name) {
+  if (!event || !event.name || Array.isArray(event.name)) {
```

Comment:
why the additional check ` || Array.isArray(event.name)`?

Reply:
> why the additional check ` || Array.isArray(event.name)`?

Applying types to `WebhookEvent` it determines the format can be an array:
https://github.com/dominguezcelada/webhooks.js/blob/12b6714cb2693fa651b67462f69cc2789099a1c2/src/index.d.ts#L1-L6

because of `http.d.ts` (https://github.com/DefinitelyTyped/DefinitelyTyped/blob/069f404e6611098c620a265d05782db36efb4fc6/types/node/v12/http.d.ts#L63)

So, to avoid type errors in the code I had to apply this extra check before continuing the execution:

![image](https://user-images.githubusercontent.com/2574275/79704280-f84ef300-82b0-11ea-9af8-14b0f607ce04.png)

Reply:
This is a breaking change of the API. `.on()` accepts both a string and an array of strings. We have to find a way to handle that correctly ... I know it's going to be more tricky with the type definitions, but we collaborate on that

Reply:
> This is a breaking change of the API. `.on()` accepts both a string and an array of strings. We have to find a way to handle that correctly ... I know it's going to be more tricky with the type definitions, but we collaborate on that

Sure! I thought it was not on octokit's end this type, and we only had to cover single string. Let me try removing this check and sort out how to manage this double type of entry :)

Reply:
Pushed changes for that